### PR TITLE
[proto] Add extra_outputs to go_proto_compile

### LINKED ISSUE #3630: Allow extra_outputs on go_proto_compile
### What version of rules_go are you using?

0.40.0

### What version of gazelle are you using?

0.28.0

### What version of Bazel are you using?

 6.3.0rc1

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?
MacOS M1 Ventura and Linux AMD

### What did you do?

See below. 

### What did you expect to see?

We have a custom plugin to generate enums .pb.go files from .proto files. For `example.proto`, we expect both
`example_dbenum.pb.go` and `example.pb.go` to be generated as Bazel provider outputs. Currently, only files with the same name and `.pb.go` file extensions are allowed. I'd like to propose adding attributes to allow outputting more w/ custom proto plugins.

### PR DESCRIPTION
**Pull request by tingilee**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

**Which issues(s) does this PR fix?**

Fixes #3630

**Other notes for review**

### CODE REVIEW

File: proto/compiler.bzl [hunk 1]
```diff
@@ -93,6 +98,17 @@ def go_proto_compile(go, compiler, protos, imports, importpath):
                 ext = compiler.internal.suffix,
             )
             go_srcs.append(out)
+
+            if extra_outputs.get(src.basename):
```
Comment (fmeum) [inline proto/compiler.bzl:102]: You can assign this to a variable so that you don't need to `get` it again in the next line.

File: proto/def.bzl [hunk 2]
```diff
@@ -165,6 +166,9 @@ go_proto_library = rule(
             providers = [GoLibrary],
             aspects = [_go_proto_aspect],
         ),
+        "extra_outputs": attr.string_dict(
+            doc = "Extra output files to be included in the expected generated output",
```
Comment (fmeum) [inline proto/def.bzl:170]: Same as for the other doc, please explain the role of keys and values.

**Review by fmeum — COMMENTED** [review]
Tests in `tests/core/go_proto_library` would be highly appreciated. There are already a number of tests there that you can use as a reference, let me know if you need further support.

**Comment by tingilee** [comment]
Abandoning this in favor of https://github.com/bazelbuild/rules_go/pull/3650

**Pull request closed without merging**